AddressDataType/ConcatenationHelper.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Linq;
+
+namespace InternationalAddress;
+
+public static class ConcatenationHelper
+{
+    public static string JoinNotNullAndNotWhiteSpace(string separator, params string[] items) =>
+        string.Join(separator, items.Where(item => string.IsNullOrWhiteSpace(item)));
```

Comment:
```suggestion
        string.Join(separator, items.Where(item => !string.IsNullOrWhiteSpace(item)));
```

You're joining emptiness.

Reply:
Thanks!